### ISSUE TITLE
If Response's status is undefined and headers is set, HTTP status 10 instead of 200

### ISSUE THREAD

**Issue by SaltyAom** [oven-sh/bun#1523]
### What version of Bun is running?

0.2.2

### What platform is your computer?

Darwin 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000 arm64 arm

### What steps can reproduce the bug?

Normally, returning Response from `Bun.serve` without setting status would return `HTTP 200`
<img width="1512" alt="Screenshot 2565-11-18 at 20 42 48" src="https://user-images.githubusercontent.com/35027979/202719185-4e778b64-02eb-4ff4-8bba-20b6d54e2542.png">

However, if Response is returned with `headers` and status code is `undefined`, status code will be 10 instead of 200.
<img width="1512" alt="Screenshot 2565-11-18 at 20 41 06" src="https://user-images.githubusercontent.com/35027979/202718752-c743e3d9-f4ff-4250-8b7c-7a73f49b2f7e.png">

```typescript
Bun.serve({
    port: 3000,
    fetch() {
        return new Response("hi", {
            status: undefined,
            headers: {
                Server: 'Bun'
            }
        })
    }
})
```

This doesn't occur if either one of `headers`, or `status` is missing or both is missing, it would still return 200 as expected.

If headers is presented, and no `status` is set, 200 is returned.
<img width="1512" alt="Screenshot 2565-11-18 at 20 41 36" src="https://user-images.githubusercontent.com/35027979/202719072-8e2582b1-a90b-461b-9bc0-e83e3b744ba5.png">

If `status` is undefined, and no `headers` is set, 200 is returned.
<img width="1512" alt="Screenshot 2565-11-18 at 20 42 11" src="https://user-images.githubusercontent.com/35027979/202719152-1c9decd2-6549-4b47-a6b0-b02e5dd465a7.png">

### How often does it reproduce? Is there a required condition?

If Response status is `undefined` and `headers` are available, status will be 10 instead of 200.

### What is the expected behavior?

Response status is should be 200, as leaving both or either one returns 200, settings `status` as undefined with `header` should also follow the same convention.

### What do you see instead?

Status response is 10 instead of 200

### Additional information

This case would probably be encountered by an HTTP library author I guess, as nobody would just intentionally put `status` as `undefined`.

Mostly I can set the default status as 200 instead of undefined, so it might not be that critical.

**Comment by Jarred-Sumner** [comment]
lol this is a weird one. Thanks for flagging.

**Comment by Electroid** [comment]
This is fixed as of Bun v0.5.0.
